src/mdns/mdns.hpp

```diff
@@ -561,6 +563,19 @@ class Publisher : private NonCopyable
     std::map<std::string, Timepoint> mHostResolutionBeginTime;
 
     otbr::MdnsTelemetryInfo mTelemetryInfo{};
+
+public:
+    const ServiceRegistration *FindServiceRegistrationByType(const std::string &aType) const
+    {
+        for (const auto &serviceReg : mServiceRegistrations)
+        {
+            if (serviceReg.second->mType == aType)
+            {
+                return serviceReg.second.get();
+            }
+        }
+        return nullptr;
+    }
```

Comment:
1. When is this method used?
2. Does it have to be `public`?
3. Move the implementation to `mdns.cpp`.

Reply:
Removed this part for the PR. I may add it back in a future one.